internal/kube/client.go

```diff
@@ -93,20 +93,19 @@ type ResolvedServicePort struct {
 
 // ResolveServicePorts converts named ports into their true
 // format. TargetPort's that have are named become their integer equivalents
-func ResolveServicePorts(ctx context.Context, k kubernetes.Interface,
-	s *corev1.Service) ([]ResolvedServicePort, bool, error) {
-	store := kevents.GlobalCache.GetStore(&corev1.Endpoints{})
+func ResolveServicePorts(s *corev1.Service) ([]ResolvedServicePort, bool, error) {
+	store := kevents.GlobalCache.Core().V1().Endpoints().Informer().GetStore()
 	if store == nil {
 		return nil, false, fmt.Errorf("endpoints store was empty")
 	}
 
-	obj, exists, err := store.GetByKey(s.Namespace + "/" + s.Name)
+	obj, _, err := store.GetByKey(s.Namespace + "/" + s.Name)
 	if err != nil {
 		return nil, false, errors.Wrap(err, "failed to get endpoints")
 	}
 
-	e := obj.(*corev1.Endpoints)
-	if !exists || len(e.Subsets) == 0 {
+	e, ok := obj.(*corev1.Endpoints)
```

Comment:
Can it ever not being `*corev1.Endpoints`? It's probably better to check the type here, but seems like a case that can't happen. That being said, totally happy to be overly cautious!

Reply:
I think I did that because I'm ignoring the "exists" argument from the `Get` call so I think it could be that obj is actually `nil`.